ts/main.ts

```diff
@@ -256,3 +266,44 @@ $confirmDeleteBtn.addEventListener('click', () => {
   viewSwap('entries');
   resetEntryForm();
 });
+
+$searchForm.addEventListener('submit', (event: Event) => {
+  event.preventDefault();
+
+  const searchTerm = $searchInput.value.toLowerCase();
+
+  if (!searchTerm) return;
+
+  $searchInput.className = 'with-clear';
+  $clearSearchBtn.className = '';
+
+  for (let i = 0; i < data.entries.length; i++) {
+    const $currentLi = document.querySelector(
+      `li[data-entry-id="${data.entries[i].entryId}"]`,
+    );
+    if (!$currentLi) throw new Error('$hideLi query failed');
+    if (
```

Comment:
Rather than querying for each individual `li` at each iteration of the loop, you should query for _all_ of the `li`s and then loop through the _node list_ of `li`s. This will be more efficient due to only using one query for all of them rather than an unknown number of queries for each. It may not be too slow with only a few entries but what if there were 50-100 entries?